rero_ils/modules/documents/dojson/contrib/marc21tojson_kul/model.py

```diff
@@ -0,0 +1,1963 @@
+    'understanding_level': re.compile(r'^target_understanding'),
+    'school_level': re.compile(r'^target_school'),
+    'filmage_ch': re.compile(r'^from the age of')
+}
+
+marc21 = ReroIlsMarc21Overdo()
+
+
+def build_place():
+    """Build place data for provisionActivity."""
+    place = {}
+    if marc21.cantons:
+        place['canton'] = marc21.cantons[0]
+    if marc21.country:
+        place['country'] = marc21.country
+    if place:
+        place['type'] = 'bf:Place'
+    if marc21.links_from_752:
+        place['identifyBy'] = marc21.links_from_752[0]
+    return place
+
+
+@marc21.over('issuance', 'leader')
+@utils.ignore_value
+def marc21_to_type_and_issuance(self, key, value):
+    """Get document type, content/Media/Carrier type and mode of issuance."""
+    if marc21.content_media_carrier_type:
+        self['contentMediaCarrier'] = marc21.content_media_carrier_type
+    if marc21.langs_from_041_h:
+        self['originalLanguage'] = marc21.langs_from_041_h
+    if marc21.admin_meta_data:
+        self['adminMetadata'] = marc21.admin_meta_data
+    # if adminMetadata/descriptionLanguage contains
+    # "Language of cataloging varies", remove it !
+    try:
+        if self['adminMetadata']['descriptionLanguage'] == \
+                'Language of cataloging varies':
+            self['adminMetadata']['descriptionLanguage'] = 'und'
+    except Exception:
+        self['adminMetadata']['descriptionLanguage'] = 'und'
+    main_type = _ISSUANCE_MAIN_TYPE_PER_BIB_LEVEL.get(
+        marc21.bib_level, 'rdami:1001')
+    sub_type = 'NOT_DEFINED'
+    error = False
+    if marc21.bib_level == 'm':
+        if marc21.is_top_level_record:
+            main_type = 'rdami:1002'
+            sub_type = 'set'
+        else:
+            sub_type = 'materialUnit'
+    else:
+        if marc21.bib_level in _ISSUANCE_SUBTYPE_PER_BIB_LEVEL:
+            sub_type = _ISSUANCE_SUBTYPE_PER_BIB_LEVEL[marc21.bib_level]
+        elif marc21.serial_type in _ISSUANCE_SUBTYPE_PER_SERIAL_TYPE:
+            sub_type = _ISSUANCE_SUBTYPE_PER_SERIAL_TYPE[marc21.serial_type]
+    if main_type == 'rdami:1001':
+        if sub_type not in [
+                'article', 'materialUnit', 'privateFile', 'privateSubfile'
+        ]:
+            error = True
+            sub_type = 'materialUnit'
+    elif main_type == 'rdami:1002':
+        if sub_type not in [
+                'set', 'partIndependentTitle', 'partDependantTitle'
+        ]:
+            error = True
+            sub_type = 'set'
+    elif main_type == 'rdami:1003':
+        if sub_type not in [
+            'serialInSerial', 'monographicSeries', 'periodical'
+        ]:
+            error = True
+            sub_type = 'periodical'
+    elif main_type == 'rdami:1004':
+        if sub_type not in ['updatingWebsite', 'updatingLoose-leaf']:
+            error = True
+            sub_type = 'updatingWebsite'
+    if error:
+        error_print('WARNING ISSUANCE:', marc21.bib_id, marc21.rero_id,
```

Comment:
Warning or Error?

Reply:
error_print is a simple logger. (It's the same code in marc21tojson).